lib/rucio/core/replica.py

```diff
@@ -1870,7 +1950,8 @@ def __cleanup_after_replica_deletion(scope_name_temp_table, scope_name_temp_tabl
                                   models.DataIdentifierAssociation.name == parent_name))))
 
         if did_associations_to_remove:
-            session.query(association_temp_table).delete()
+            stmt = delete(association_temp_table)
+            session.execute(stmt)
             session.execute(insert(association_temp_table), [a._asdict() for a in did_associations_to_remove])
```

Comment:
Can we also break this into a statement/execution here?

Reply:
Changed to:

```
            values = [a._asdict() for a in did_associations_to_remove]
            stmt = insert(association_temp_table)
            session.execute(stmt, values)
```